modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormAdminActionDropdownItemsProviderTest.java

```diff
@@ -0,0 +1,567 @@
+import org.mockito.Mockito;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Carolina Barbosa
+ */
+@PrepareForTest(PortletProviderUtil.class)
+@RunWith(PowerMockRunner.class)
+public class DDMFormAdminActionDropdownItemsProviderTest extends PowerMockito {
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_setUpJSONFactoryUtil();
+		_setUpLanguageUtil();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_setUpDDMFormAdminActionDropdownItemsProvider(true);
+		_setUpPortletProviderUtil();
+	}
+
+	@Test
+	public void testGetActionDropdownItems() {
+		List<DropdownItem> dropdownItems =
+			_ddmFormAdminActionDropdownItemsProvider.getActionDropdownItems();
+
+		List<DropdownItem> dropdownGroupItems = _getDropdownGroupItems(
+			dropdownItems, 0);
+
+		Assert.assertEquals(
+			dropdownGroupItems.toString(), 3, dropdownGroupItems.size());
+
+		_assertActionDropdownItemEdit(dropdownGroupItems.get(0));
+		_assertActionDropdownItemViewEntries(dropdownGroupItems.get(1));
+		_assertActionDropdownItemShare(dropdownGroupItems.get(2));
+
+		dropdownGroupItems = _getDropdownGroupItems(dropdownItems, 1);
+
+		Assert.assertEquals(
+			dropdownGroupItems.toString(), 2, dropdownGroupItems.size());
+
+		_assertActionDropdownItemDuplicate(dropdownGroupItems.get(0));
+		_assertActionDropdownItemExport(dropdownGroupItems.get(1));
+
+		dropdownGroupItems = _getDropdownGroupItems(dropdownItems, 2);
+
+		Assert.assertEquals(
+			dropdownGroupItems.toString(), 1, dropdownGroupItems.size());
+
+		_assertActionDropdownItemPermissions(dropdownGroupItems.get(0));
+
+		dropdownGroupItems = _getDropdownGroupItems(dropdownItems, 3);
+
+		Assert.assertEquals(
+			dropdownGroupItems.toString(), 1, dropdownGroupItems.size());
+
+		_assertActionDropdownItemDelete(dropdownGroupItems.get(0));
+	}
+
+	@Test
+	public void testGetActionDropdownItemsNotShowingDelete() throws Exception {
+		when(
+			_formInstancePermissionCheckerHelper.isShowDeleteIcon(
+				_ddmFormInstance)
+		).thenReturn(
+			false
+		);
+
+		List<DropdownItem> dropdownGroupItems = _getDropdownGroupItems(
+			_ddmFormAdminActionDropdownItemsProvider.getActionDropdownItems(),
+			3);
+
+		Assert.assertEquals(
+			dropdownGroupItems.toString(), 0, dropdownGroupItems.size());
+	}
+
+	@Test
+	public void testGetActionDropdownItemsNotShowingDuplicate()
+		throws Exception {
+
+		when(
+			_formInstancePermissionCheckerHelper.isShowDuplicateIcon()
+		).thenReturn(
+			false
+		);
+
+		List<DropdownItem> dropdownGroupItems = _getDropdownGroupItems(
+			_ddmFormAdminActionDropdownItemsProvider.getActionDropdownItems(),
+			1);
+
+		Assert.assertEquals(
+			dropdownGroupItems.toString(), 1, dropdownGroupItems.size());
+
+		_assertActionDropdownItemExport(dropdownGroupItems.get(0));
```

Comment:
As offline discussed, let's get rid of the assertion since we're guaranteeing the order and correct implementation of each item in the main test.

Reply:
Done!